scripts/verify-jenkins-on-openshift.sh

```diff
@@ -0,0 +1,145 @@
+#!/bin/bash
+
+# This script deploys the jenkins pod with provided image (from CPaas) on an 
+# OpenShift cluster & executes verify_jenkins.sh within the newly deployed pod.
+# 
+# Prerequisites:
+# Ensure a login to OpenShift cluster, either via user:password, token
+# OR
+# KUBECONFIG env var refers to a valid kubeconfig
+# 
+# Usage: ./verify-jenkins-on-openshift.sh [-i <jenkins_image>] [-s <commit_sha>]
+# 
+
+set -o pipefail
+
+# returns the details about command usage
+usage() { 
+    printf "Usage: ./verify-jenkins-on-openshift.sh [-i <jenkins_image>] [-s <commit_sha>]\n"
+    printf "Pre-requisite: Ensure that a user is already logged in the cluster OR have a valid KUBECONFIG populated\n" 
+    exit 1; 
+}
+
+# Assigns the variable values as per the flags passed.
+while getopts "i:s:*:" flag
+do
+    case "${flag}" in
+        i) JENKINS_IMAGE=${OPTARG};;
+        s) COMMIT_SHA=${OPTARG};;
+        *) usage
+    esac
+done
+
+# Verifies & sets the required values for variables.
+# 
+vars(){   
+
+    if [ -z "${JENKINS_IMAGE}" ] || [ -z "${COMMIT_SHA}" ]
+    then
+        usage
+    fi
+
+    # The image pushed will have the respective commit's sha assigned as the tag
+    QUAY_IMAGE="quay.io/pipeline-integrations/openshift-ose-jenkins:${COMMIT_SHA}"
+}
+
+# deploy_on_openshift()
+# Deploys the pod with the provided CPaas image, 
+# 
+deploy_on_openshift(){
+
+    # ensure oc exists
+    oc version > /dev/null
+    if [ $? != 0 ]
+    then
+        printf "\n>ERR || Please install 'oc' client to continue. https://mirror.openshift.com/pub/openshift-v4/clients/ocp/latest/\n"
+        printf ">ERR || And/OR ensure that the user has been logged in to the cluster.\n"
+        exit 1
+    fi
+
+    # if oc exists, continue
+    oc delete project test-jenkins-latest
+    oc new-project test-jenkins-latest
+    oc import-image --confirm jenkins-image --from="${QUAY_IMAGE}"
+    oc new-app jenkins-ephemeral -p NAMESPACE="$(oc project -q)" -p JENKINS_IMAGE_STREAM_TAG=jenkins-image:latest
+
+    # wait for jenkins pod to switch to 'Ready' & 'Running` state:
+    sleep 10
+    printf "\n-------------------------------------------------------\n>INFO || Waiting for pod to be ready\n"
+    OUTPUT=$(oc wait pod --for=condition=Ready -l name=jenkins,deploymentconfig=jenkins --timeout=120s)
+
+    if [ -z "${OUTPUT}" ]
+    then
+        exit 1
+    else
+        IFS=" " read -ra ELEMENTS <<< "${OUTPUT}"
+        IFS="/" read -r POD POD_NAME <<< "${ELEMENTS[0]}"
+    fi
+
+    printf "\n-------------------------------------------------------\n>INFO || Pod %s is now available. Any additional tests can be performed against it.\n" "${POD_NAME}"
```

Comment:
Can we expand this comment to mention needing to run the `make e2e` command and verify that all tests pass?

Reply:
Did not change it yet, as still contemplating what could be the best way to use this pod. Let me know your thoughts :)